baselines/dasha/README.md

```diff
@@ -0,0 +1,118 @@
+
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# Install the base Poetry environment
+# By default, it is assumed that the user has Python 3.8 and CUDA 11.8. 
+# If you have a different setup, then change the "torch" and "torchvision" lines in [tool.poetry.dependencies].
+poetry install
+
+# Activate the environment
+poetry shell
+```
+
+
+## Running the Experiments
+
+To run this FedProx with MNIST baseline, first ensure you have activated your Poetry environment (execute `poetry shell` from this directory), then:
+
+```bash
+python -m dasha.main # this will run using the default settings in `dasha/conf`
+
+# you can override settings directly from the command line
+# The following commands create a new directory for the dataset, and then it runs an experiment with the step size 0.5.
+# Instead of the full, non-compressed vectors, each node sends a compressed vector with only 10 coordinates.
+mkdir -p $HOME/.flower/dataset
+python -m dasha.main dataset.path_to_dataset=$HOME/.flower/dataset method.strategy.step_size=0.5 compressor.number_of_coordinates=10
+
+# if you run this baseline with a larger model, you might want to use the GPU (not used by default).
+python -m dasha.main method.client.device=cuda
+```
+
+To run using MARINA by Gorbunov et al. (2020):
+```bash
+python -m dasha.main method=marina
+```
+
+
+## Expected Results
+### Small-Scale Experiments
+
+With the following command we run both DASHA and MARINA methods while iterating through different values of `step size`. Other parameters are the same as in the original paper. In the following command, we also ask the clients to send the full gradients to evaluate the norm of gradients metric.
+
+```bash
+mkdir -p $HOME/.flower/dataset
+# Run experiments
+python -m dasha.main --multirun method=dasha,marina compressor.number_of_coordinates=10 method.strategy.step_size=0.25,0.5,1.0 method.client.send_gradient=true dataset.path_to_dataset=$HOME/.flower/dataset save_path=$HOME/.flower/results
+# Plot results
+python -m dasha.plot --input_paths $HOME/.flower/results --output_path $HOME/.flower/plot.png --metric squared_gradient_norm
+```
+
+The above commands would generate results that you can plot and would look like:
+
+![](docs/plot.png)
+
+### Large-Scale Experiments
+
+In the following experiments, we compare the performance of DASHA and MARINA on the CIFAR10 dataset.
+```bash
+mkdir -p $HOME/.flower/dataset
+# Run experiments
+python dasha/main.py dataset.path_to_dataset=$HOME/.flower/dataset method.strategy.step_size=0.01 method=stochastic_dasha num_rounds=10000 compressor.number_of_coordinates=2000000 model=resnet_18_with_logistic_loss method.client.strict_load=false dataset=cifar10 method.client.device=cuda method.client.evaluate_accuracy=true local_address=localhost:8001 method.client.mega_batch_size=16
```

Comment:
maybe replace with `python -m dasha.main <...>` for consistency? (also the command below)